test/main/async.js

```diff
@@ -164,9 +187,34 @@ QUnit.test( "fails if callback is called more than once in test", function( asse
 		var overDone = assert.async();
 		overDone();
 		overDone();
-	}, new RegExp( "Called the callback returned from `assert.async` more than once" ) );
+	}, new RegExp( "Too many calls to the `assert.async` callback" ) );
+
+	done();
+});
+
+QUnit.test( "fails if callback is called more than callback call count", function( assert ) {
+
+	// Having an outer async flow in this test avoids the need to manually modify QUnit internals
+	// in order to avoid post-`done` assertions causing additional failures
+	var done = assert.async();
+
+	assert.expect( 1 );
+
+	// Duck-punch to force an Error to be thrown instead of a `pushFailure` call
+	assert.test.pushFailure = function( msg ) {
+		throw new Error( msg );
+	};
+
+	assert.throws(function() {
+		var overDone = assert.async( 3 );
+		overDone();
+		overDone();
+		overDone();
+		overDone();
+	}, new RegExp( "Too many calls to the `assert.async` callback" ) );
```

Comment:
only the last overDone should be inside this function.

```js
	var overDone = assert.async( 3 );
	overDone();
	overDone();
	overDone();
	assert.throws(function() {		
		overDone();
	}, new RegExp( "Too many calls to the `assert.async` callback" ) );
```

This reveals a similar issue on a previous existing assertion: https://github.com/jquery/qunit/pull/863/files#diff-fbe33c9a2379400e62e9f4120ec9c024R190